support-utils/cre_astro_software_gather_logs.sh

```diff
@@ -0,0 +1,392 @@
+      echo "Gathering secrets in $NS Namespace ";kubectl get secrets > "$NS/AllPodlogs/morelogs/secrets_$NS.log" -n $NS
+      echo "Gathering sevice Status in $NS Namespace ";kubectl get svc > "$NS/AllPodlogs/morelogs/svc_$NS.log" -n $NS
+      echo "Gathering persistent volume Status in $NS Namespace ";kubectl get pvc > "$NS/AllPodlogs/morelogs/$PVC_DIR/pvc_$NS.log" -n $NS
+      echo "Gathering ingress Status in $NS Namespace ";kubectl get ingress > "$NS/AllPodlogs/morelogs/ingress_$NS.log" -n $NS
+      echo "Gathering cronjobs Status in $NS Namespace ";kubectl get cronjobs > "$NS/AllPodlogs/morelogs/cronjobs_$NS.log" -n $NS
+      echo "Gathering jobs Status in $NS Namespace ";kubectl get jobs > "$NS/AllPodlogs/morelogs/cronjobs_NS.log" -n $NS
+      echo "Number of digits to trim while getting release name";export X=$(echo $ASTRONOMER_NAMESPACE|wc -m);echo "VALUE to be increased by 1 for trimming is $X in $NS";export Y=$(($X+1)); echo "Number of digits to be trimmed while getting release name is $Y in $NS"
+      echo "Exporting Release name ";export Release_Name=$(echo $NS| cut -c $Y-)
+      echo "Your Release_Name in current namespace $NS is $Release_Name."
+echo "======================Gathering All the Deployment namespace logs in the $NS======================"
+      echo "Gathering logs of scheduler in $NS Namespace ";kubectl logs deployment/$Release_Name-scheduler -c scheduler > "$NS/AllPodlogs/morelogs/scheduler_$NS.log" -n $NS  
+      echo "Gathering logs of worker in $NS Namespace ";kubectl logs deployment/$Release_Name-worker -c worker > "$NS/AllPodlogs/morelogs/worker_$NS.log" -n $NS
+      echo "Gathering logs of webserverin $NS Namespace ";kubectl logs deployment/$Release_Name-webserver -c webserver > "$NS/AllPodlogs/morelogs/webserver$NS.log" -n $NS
+      echo "Gathering logs of triggerer in $NS Namespace ";kubectl logs deployment/$Release_Name-triggerer -c triggerer > "$NS/AllPodlogs/morelogs/triggerer $NS.log" -n $NS
+      echo "Gathering logs of pgbouncer in $NS Namespace ";kubectl logs deployment/$Release_Name-pgbouncer -c pgbouncer > "$NS/AllPodlogs/morelogs/pgbouncer _$NS.log" -n $NS    
+      echo "Gathering logs of flower  in $NS Namespace ";kubectl logs deployment/$Release_Name-flower > "$NS/AllPodlogs/morelogs/flower _$NS.log" -n $NS
+      echo "Gathering logs of statsd in $NS Namespace ";kubectl logs deployment/$Release_Name-statsd > "$NS/AllPodlogs/morelogs/statsd_$NS.log" -n $NS
+      echo "Gathering logs of redis in $NS Namespace ";kubectl logs sts/$Release_Name-redis > "$NS/AllPodlogs/morelogs/redis_$NS.log" -n $NS 
+      echo "Gathering helm history in $NS Namespace";helm history $Release_Name -n $NS > "$NS/AllPodlogs/morelogs/helm_history_$Release_Name.yaml"
+      echo "Gathering helm values from $NS Namespace";helm get values $Release_Name -o yaml -n $NS  > "$NS/AllPodlogs/morelogs/helm_values_$Release_Name.yaml"
```

Comment:
This is a lot of duplicate code that should probably be turned into a function that takes the name of the pod, the namespace, and the release name.